packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -559,6 +559,11 @@ mixin SchedulerBinding on BindingBase {
   /// Adds the given callback to the list of frame callbacks and ensures that a
   /// frame is scheduled.
   ///
+  /// If this is called during the frame's animation phase (when transient frame
+  /// callbacks are still being invoked), a new frame will be scheduled, and
+  /// `callback` will be called in the newly scheduled frame, not in the current
+  /// frame.
```

Comment:
might be worth considering ways to merge this with the previous paragraph. but if no good merged version is obvious, this if fine too imho.